index.js

```diff
@@ -410,7 +410,10 @@ exports.parseNoPatch = function (code, options) {
       ast = jsParse(code, opts);
     }
   } catch (err) {
-    if (err instanceof SyntaxError) {
+    // Any error with location information should be promoted to a code frame.
+    // Otherwise linter plugins for IDEs lose the positional information for
+    // semantic errors.
+    if (err.loc) {
```

Comment:
Though I will admit -- this is a hack (but no more than the linter itself). Without this change, it's borderline useless because you just get a red cursor at line 0:0 on semantic linting errors.